primefaces/src/main/java/org/primefaces/csp/CspState.java

```diff
@@ -53,7 +53,7 @@ public CspState(FacesContext context) {
      */
     public String getNonce() {
         if (nonce == null) {
-            if (context.isPostback()) {
+            if (context.isPostback() || context.getPartialViewContext().isAjaxRequest()) {
```

Comment:
isnt a AJAX request always a postback?

Reply:
I am not sure if this has to do with the "phase" the ExceptionListener is catching the error or something else?